strax/context.py

```diff
@@ -1078,7 +1121,7 @@ def function(arr):
             if store_first_for_others and not seen_data and len(data):
                 # Store the first value we see for the non-accumulated fields
                 for name in data.dtype.names:
-                    if name in fields:
+                    if name not in fields:
```

Comment:
Dear reviewer, I also caught a little but in accumulate which I fixed. Since it is a trivial change I did not want to open a separate PR.